cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -631,11 +635,15 @@ public void resumeLoadMonitorActivity(PauseResumeParameters parameters) {
                                                                 boolean skipHardGoalCheck,
                                                                 Pattern excludedTopics,
                                                                 boolean excludeRecentlyDemotedBrokers,
-                                                                boolean excludeRecentlyRemovedBrokers)
+                                                                boolean excludeRecentlyRemovedBrokers,
+                                                                boolean ignoreProposalCache)
       throws KafkaCruiseControlException {
     GoalOptimizer.OptimizerResult result;
     sanityCheckHardGoalPresence(goals, skipHardGoalCheck);
     List<Goal> goalsByPriority = goalsByPriority(goals);
+    if (ignoreProposalCache) {
+      goals = goalsByPriority.stream().map(Goal::name).collect(Collectors.toList());
+    }
```

Comment:
instead of doing this, can we direct put `ignoreProposalCache` in to the `if-else` statement below?
`if ((goals != null && !goals.isEmpty())...` -> `if (ignoreProposalCache || (goals != null && !goals.isEmpty())...`